scripts/azure/storage-manager.sh

```diff
@@ -0,0 +1,138 @@
+
+objectstorage_type_sc=$(echo "${sc_config}" | yq4 '.objectStorage.type' -)
+objectstorage_type_wc=$(echo "${wc_config}" | yq4 '.objectStorage.type' -)
+
+[ "$objectstorage_type_sc" != "azure" ] && log_info "Azure Storage is not enabled in service cluster"
+[ "$objectstorage_type_wc" != "azure" ] && log_info "Azure Storage is not enabled in workload cluster"
+
+if [ "$objectstorage_type_sc" != "azure" ] && [ "$objectstorage_type_wc" != "azure" ]; then
+    log_error "Azure Storage is not enabled in either cluster, aborting!"
+    exit 1
+fi
+
+[ "$objectstorage_type_sc" = "azure" ] && buckets_sc=$(echo "${sc_config}" | yq4 '.objectStorage.buckets.*' -)
+[ "$objectstorage_type_wc" = "azure" ] && buckets_wc=$(echo "${wc_config}" | yq4 '.objectStorage.buckets.*' -)
+
+CONTAINERS=$( { echo "$buckets_sc"; echo "$buckets_wc"; } | sort | uniq | tr '\n' ' ' | sed s'/.$//')
+
+function create_resource_group() {
+
+    echo "checking if resource group exists" >&2
+    GROUP_EXISTS=$(az group list --query '[].name' | awk "/${CK8S_ENVIRONMENT_NAME}-storage-resource-group/")
+    if [ "$GROUP_EXISTS" ]; then
+        echo "resource group [${CK8S_ENVIRONMENT_NAME}-storage-resource-group] already exists" >&2
+        echo "continue using this group ? (y/n)" >&2
+        read -r -n 1 cmdinput
+        case "$cmdinput" in
+        y|Y) return ;;
+        *) exit 0 ;;
+        esac
+    else
+        echo "resource group [${CK8S_ENVIRONMENT_NAME}-storage-resource-group] does not exist, creating it now" >&2
+        az group create \
+            --name "$CK8S_ENVIRONMENT_NAME"-storage-resource-group \
+            --location "${AZURE_LOCATION}" --only-show-errors
+    fi
+
+}
+
+function create_storage_account() {
+
+    echo "checking if storage account exists" >&2
+    ACCOUNT_EXISTS=$(az storage account list --query '[].name' | awk "/${CK8S_ENVIRONMENT_NAME}storageaccount/")
+    if [ "$ACCOUNT_EXISTS" ]; then
+        echo "storage account [${CK8S_ENVIRONMENT_NAME}storageaccount] already exists" >&2
+        echo "contnue using this account ? (y/n)" >&2
+        read -r -n 1 cmdinput
+        case "$cmdinput" in
+        y) return ;;
+        n) exit 0 ;;
+        esac
+    else
+        az storage account create \
+            --name "$CK8S_ENVIRONMENT_NAME"storageaccount \
+            --resource-group "$CK8S_ENVIRONMENT_NAME"-storage-resource-group \
+            --location swedencentral \
+            --sku Standard_RAGRS \
+            --kind StorageV2 \
+            --allow-blob-public-access false --only-show-errors
+    fi
+}
+
+function create_containers() {
+
+    CONTAINERS_LIST=$(az storage container list --account-name "$CK8S_ENVIRONMENT_NAME"storageaccount --query '[].name' --only-show-errors)
+
+    # shellcheck disable=SC2068
+    for container in ${CONTAINERS[@]}; do
+
+        echo "checking status of container ${container}]" >&2
+
+        CONTAINER_EXISTS=$(echo "$CONTAINERS_LIST" | awk "/${container}/")
```

Comment:
While this probably won't become a problem this could give a false positive, for example a container named `foobar` would make this script think a container named only `foo` exists.

For example:
```shell
$ CONTAINERS_LIST=$(az storage container list --account-name hanistorageaccount --query '[].name' --only-show-errors)
$ echo "$CONTAINERS_LIST" | awk "/hani-audit/"
  "hani-audit",
$ echo "$CONTAINERS_LIST" | awk "/hani-a/"
  "hani-audit",
```

Since the command outputs json, what you could do is to use `jq`, something like:
```shell
$ echo $CONTAINERS_LIST | jq '. | index("hani-sclogs")'
3
$ echo $CONTAINERS_LIST | jq '. | index("doesnotexist")'
null
```